src/err/mod.rs

```diff
@@ -178,8 +178,10 @@ impl PyErr {
     pub fn from_value(obj: &PyAny) -> PyErr {
         let state = if let Ok(obj) = obj.downcast::<PyBaseException>() {
             PyErrState::Normalized(PyErrStateNormalized {
```

Comment:
I think it would be nice to have a `PyErrState::normalized` constructor to better consolidate the `#[cfg(Py_3_12)]` attributes in the `err_state` module.